addon/mixins/adapter-fetch.js

```diff
@@ -211,44 +221,54 @@ export default Ember.Mixin.create({
   },
 
   /**
+   * @param {Object} adapter
    * @param {Object} response
-   * @param {Promise} bodyPromise
+   * @param {Object} payload
    * @param {Object} requestData
    * @override
    */
-  ajaxSuccess(response, bodyPromise, requestData) {
-    const headersObject = headersToObject(response.headers);
+  ajaxSuccess(adapter, response, payload, requestData) {
+    const returnResponse = adapter.handleResponse(
+      response.status,
+      headersToObject(response.headers),
+      payload,
+      requestData
+    );
 
-    return bodyPromise.then((body) => {
-      const returnResponse = this.handleResponse(
-        response.status,
-        headersObject,
-        body,
-        requestData
-      );
+    if (returnResponse && returnResponse.isAdapterError) {
+      return RSVP.Promise.reject(returnResponse);
+    } else {
+      return returnResponse;
+    }
+  },
 
-      if (returnResponse && returnResponse.isAdapterError) {
-        return RSVP.Promise.reject(returnResponse);
-      } else {
-        return returnResponse;
-      }
-    });
+
+/**
+ * Allows for the error to be selected from either the
+ * response object, or the response data.
+ * @param {Object} response
+ * @param {Object} responseData
+ */
+  parseFetchResponseForError(response, responseData) {
```

Comment:
my only concern here about payload, is that ember data doesn't call it payload in the ajaxSuccess method https://github.com/emberjs/data/blob/master/addon/adapters/rest.js#L1507 so i was wanting to use thier verbiage to maintain consistency. (even though they are inconsistent between thier error and success methods)